exspy/models/edsmodel.py

```diff
@@ -827,9 +853,28 @@ def calibrate_xray_lines(
         """
 
         if calibrate == "energy":
-            bound = (bound / eV2keV) * self.units_factor
-            free = self.free_xray_lines_energy
-            fix = self.fix_xray_lines_energy
+            # For energy calibration, reset line energies to reference values from the database
+            # and keep them fixed, rather than optimizing them through fitting
+            target_lines = (
+                xray_lines if xray_lines != "all" else [c.name for c in self.xray_lines]
+            )
+            for xray_line in target_lines:
+                try:
```

Comment:
Why is this `try: except:` block necessary? If some line from xraydb are missing, it will be caught in `get_xray_line_energy_xraydb`?